addons/sale_stock/models/sale_order.py

```diff
@@ -137,14 +137,15 @@ def _onchange_product_id_check_availability(self):
             return {}
         if self.product_id.type == 'product':
             precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
+            virtual_available = self.product_id.with_context(warehouse=self.order_id.warehouse_id.id).virtual_available
             product_qty = self.product_uom._compute_quantity(self.product_uom_qty, self.product_id.uom_id)
-            if float_compare(self.product_id.virtual_available, product_qty, precision_digits=precision) == -1:
+            if float_compare(virtual_available, product_qty, precision_digits=precision) == -1:
                 is_available = self._check_routing()
                 if not is_available:
                     warning_mess = {
                         'title': _('Not enough inventory!'),
-                        'message' : _('You plan to sell %s %s but you only have %s %s available!\nThe stock on hand is %s %s.') % \
-                            (self.product_uom_qty, self.product_uom.name, self.product_id.virtual_available, self.product_id.uom_id.name, self.product_id.qty_available, self.product_id.uom_id.name)
+                        'message' : _('You plan to sell %s %s but you only have %s %s available in %s.\nThere are %s %s available in all warehouses.') % \
+                            (self.product_uom_qty, self.product_uom.name, virtual_available, self.product_id.uom_id.name, self.order_id.warehouse_id.name, self.product_id.virtual_available, self.product_id.uom_id.name)
```

Comment:
i dont think it's good idea to change this message. if multi warehouse is not enabled then same warning raised  that make confuse.